File: demo-drag-drop.py
# Demo Python File for Drag & Drop Testing
# Try dragging this file into the chat input area!

import json
import datetime
from typing import List, Dict, Optional

class MessageProcessor:
    def __init__(self, room_id: str):
        self.room_id = room_id
        self.processed_count = 0
    
    def process_message(self, message: Dict) -> Optional[Dict]:
        """Process and validate a chat message."""
        try:
            # Validate required fields
            required_fields = ['content', 'username', 'timestamp']
            if not all(field in message for field in required_fields):
                return None
            
            # Clean and format content
            content = message['content'].strip()
            if not content:
                return None
            
            # Add metadata
            processed_message = {
                **message,
                'content': content,
                'room_id': self.room_id,
                'processed_at': datetime.datetime.now().isoformat(),
                'word_count': len(content.split())
            }
            
            self.processed_count += 1
            return processed_message
            
        except Exception as e:
            print(f"Error processing message: {e}")
            return None
    
    def batch_process(self, messages: List[Dict]) -> List[Dict]:
        """Process multiple messages at once."""
        processed = []
        for msg in messages:
            result = self.process_message(msg)
            if result:
                processed.append(result)
        
        print(f"✅ Processed {len(processed)} out of {len(messages)} messages")
        return processed

# Usage example
if __name__ == "__main__":
    processor = MessageProcessor("room-123")
    
    sample_messages = [
        {
            "content": "Hello world!",
            "username": "Alice",
            "timestamp": "2024-01-15T10:30:00Z"
        },
        {
            "content": "How's everyone doing?",
            "username": "Bob", 
            "timestamp": "2024-01-15T10:31:00Z"
        }
    ]
    
    results = processor.batch_process(sample_messages)
    print(json.dumps(results, indent=2))

File: demo-drag-drop.js
// Demo JavaScript File for Drag & Drop Testing
// Try dragging this file into the chat input area!

class ChatDemo {
  constructor(username) {
    this.username = username;
    this.messages = [];
  }

  sendMessage(content) {
    const message = {
      id: Date.now(),
      username: this.username,
      content: content,
      timestamp: new Date().toISOString()
    };
    
    this.messages.push(message);
    console.log(`📤 ${this.username} sent: ${content}`);
    return message;
  }

  async fetchMessages() {
    try {
      const response = await fetch('/api/messages');
      const data = await response.json();
      return data.messages || [];
    } catch (error) {
      console.error('Failed to fetch messages:', error);
      return [];
    }
  }

  formatTimestamp(timestamp) {
    return new Date(timestamp).toLocaleTimeString();
  }
}

// Usage example
const chat = new ChatDemo('CodeTester');
chat.sendMessage('Hello from drag & drop!');